docs/grpc/c#.md

```diff
@@ -0,0 +1,219 @@
+
+After installing these, use the code below to set up a channel and client to connect to your `lnd` node:
+
+```c#
+
+using Grpc.Core;
+using Lnrpc;
+...
+
+// Due to updated ECDSA generated tls.cert we need to let gprc know that
+// we need to use that cipher suite otherwise there will be a handshake
+// error when we communicate with the lnd rpc server.
+System.Environment.SetEnvironmentVariable("GRPC_SSL_CIPHER_SUITES", "HIGH+ECDSA");
+            
+// Lnd cert is at AppData/Local/Lnd/tls.cert on Windows
+// ~/.lnd/tls.cert on Linux and ~/Library/Application Support/Lnd/tls.cert on Mac
+var cert = File.ReadAllText(<Tls_Cert_Location>);
+
+var sslCreds = new SslCredentials(cert);
+var channel = new Grpc.Core.Channel("localhost:10009", sslCreds);
+var client = new Lnrpc.Lightning.LightningClient(channel);
+
+```
+
+### Examples
+
+Let's walk through some examples of C# `gRPC` clients. These examples assume that you have at least two `lnd` nodes running, the RPC location of one of which is at the default `localhost:10009`, with an open channel between the two nodes.
+
+#### Simple RPC
+
+```c#
+// Retrieve and display the wallet balance
+// Use "WalletBalanceAsync" if in async context
+var response = client.WalletBalance(new WalletBalanceRequest());
+Console.WriteLine(response);
+```
+
+#### Response-streaming RPC
+
+```c#
+var request = new InvoiceSubscription();
+using (var call = client.SubscribeInvoices(request))
+{
+    while (await call.ResponseStream.MoveNext())
+    {
+        var invoice = call.ResponseStream.Current;
+        Console.WriteLine(invoice.ToString());
+    }
+}
+```
+
+Now, create an invoice for your node at `localhost:10009`and send a payment to it from another node.
+```bash
+$ lncli addinvoice --amt=100
+{
+    "r_hash": <R_HASH>,
+    "pay_req": <PAY_REQ>
+}
+$ lncli sendpayment --pay_req=<PAY_REQ>
+```
+
+Your console should now display the details of the recently satisfied invoice.
+
+#### Bidirectional-streaming RPC
+
+```c#
+using (var call = client.SendPayment())
+{
+    var responseReaderTask = Task.Run(async () =>
+    {
+        while (await call.ResponseStream.MoveNext())
+        {
+            var payment = call.ResponseStream.Current;
+            Console.WriteLine(payment.ToString());
+        }
+    });
+
+    foreach (SendRequest sendRequest in SendPayment())
+    {
+        await call.RequestStream.WriteAsync(sendRequest);
+    }
+    await call.RequestStream.CompleteAsync();
+    await responseReaderTask;
+}
+
+
+IEnumerable<SendRequest> SendPayment()
+{
+    while (true)
+    {
+        SendRequest request = new SendRequest() {
+            DestString = "<DEST_PUB_KEY>",
+            Amt = 100,
+            PaymentHashString = "<R_HASH>",
+            FinalCltvDelta = "<CLTV_DELTA>"
+        };
+        yield return request;
+        System.Threading.Thread.Sleep(2000);
+    }
+}
+```
+This example will send a payment of 100 satoshis every 2 seconds.
```

Comment:
The idea is clear. However the example will not work completely as expected because `lnd` is currently checking for invoices that are already paid. So only the first payment succeeds.